hopp/simulation/technologies/steel/enthalpy_functions.py

```diff
@@ -59,36 +78,59 @@ def h2o_enthalpy(T):
     G=223.3967 
     H=-241.8264 
     H_t=(A*t +(B*t*t)/2 +(C*t*t*t)/3 + (D*t*t*t*t)/4-(E/t)+F-H)/mol_weight_H2O 
+
     return H_t 
 
-def fe_enthalpy_1(T):#298-1809 K
-    mol_weight_fe=55.845 #in grams
-    t=T/1000
-    A=23.97449
-    B=8.367750
-    C=0.000277
-    D=-0.000088
-    E=-0.000005
-    F=0.268027
-    G=62.06336
-    H=7.788015    
-    H_t=(A*t +(B*t*t)/2 +(C*t*t*t)/3 + (D*t*t*t*t)/4-(E/t)+F-H)/mol_weight_fe
-    return H_t
-def fe_enthalpy_2(T):#1809 - 3133K
+
+
+def fe_enthalpy(T):#298-1809 K
     mol_weight_fe=55.845 #in grams
-    t=T/1000
-    A=46.02400
-    B=-1.88467*10**(-8)
-    C=6.094750*10**(-9)
-    D=-6.640301*10**(-10)
-    E=-0.8246121*10**(-9)
-    F=-10.80543
-    G=72.54094
-    H=12.39052  
-    H_t=(A*t +(B*t*t)/2 +(C*t*t*t)/3 + (D*t*t*t*t)/4-(E/t)+F-H)/mol_weight_fe
+
+    if T < 298 or T > 3133:
+        raise ValueError(f"Inputted temperatute {T} for iron is out of range of 298-3133 K")
+
+    if T < 1809: #fe_1 298-1809 K
+        t=T/1000
+        A=23.97449
+        B=8.367750
+        C=0.000277
+        D=-0.000088
+        E=-0.000005
+        F=0.268027
+        G=62.06336
+        H=7.788015    
+        H_t=(A*t +(B*t*t)/2 +(C*t*t*t)/3 + (D*t*t*t*t)/4-(E/t)+F-H)/mol_weight_fe
+    
+    if T >= 1809: #fe_2 1809 - 3133K
+        t=T/1000
+        A=46.02400
+        B=-1.88467*10**(-8)
+        C=6.094750*10**(-9)
+        D=-6.640301*10**(-10)
+        E=-0.8246121*10**(-9)
+        F=-10.80543
+        G=72.54094
+        H=12.39052  
+        H_t=(A*t +(B*t*t)/2 +(C*t*t*t)/3 + (D*t*t*t*t)/4-(E/t)+F-H)/mol_weight_fe
+
     return H_t
+
+#def fe_enthalpy_2(T):#1809 - 3133K
+    #mol_weight_fe=55.845 #in grams
+    #t=T/1000
+    #A=46.02400
+    #B=-1.88467*10**(-8)
+    #C=6.094750*10**(-9)
+    #D=-6.640301*10**(-10)
+    #E=-0.8246121*10**(-9)
+    #F=-10.80543
+    #G=72.54094
+    #H=12.39052  
+    #H_t=(A*t +(B*t*t)/2 +(C*t*t*t)/3 + (D*t*t*t*t)/4-(E/t)+F-H)/mol_weight_fe
+    #return H_t
+
 def feo_enthalpy(T):
-    mol_weight_feo=71.844 #in grams
+    mol_weight_feo=71.844 #in grams 298-1650
```

Comment:
it looks like this and other value ranges should have been error checks on temperature, not in the comment on molecular weight